ShellPkg/Application/Shell/ShellParametersProtocol.c

```diff
@@ -1307,6 +1326,10 @@ UpdateStdInStdOutStdErr (
             TempHandle = CreateFileInterfaceFile (TempHandle, FALSE);
           }
 
+          if (TempHandle == NULL) {
+            return EFI_OUT_OF_RESOURCES;

```

Comment:
I think `CommandLineCopy` (in the function) should be freed.

It seems the functions in the ShellPkg are really long and prone to this kind of issue. I think it would help for the review to split this patch in smaller bits. Like one patch for the modifications in `ShellPkg/Library/UefiShellDebug1CommandsLib/`, one for `ShellPkg/Library/UefiShellDriver1CommandsLib/`,  etc.

Reply:
Agreed on freeing `CommandLineCopy`, updated. Also agreed on the length and complexity of functions in ShellPkg :). I will split this into one commit per component.